File: aosp/src/apps/audio_equalizer/app/src/main/java/com/leandromendes/audioequalizer/util/ProfileListUtils.java
package com.leandromendes.audioequalizer.util;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.leandromendes.audioequalizer.R;
import com.leandromendes.audioequalizer.data.model.EqualizerProfile;

import java.util.List;

public class ProfileListUtils extends ArrayAdapter<EqualizerProfile> {
    private final Context context;
    private List<EqualizerProfile> profiles;

    public ProfileListUtils(Context context, List<EqualizerProfile> profiles) {
        super(context, android.R.layout.simple_list_item_1, profiles);
        this.context = context;
        this.profiles = profiles;
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View rowView = inflater.inflate(R.layout.item_profile, parent, false);

        TextView txtNome = rowView.findViewById(R.id.profileName);
        EqualizerProfile profile = profiles.get(position);
        txtNome.setText(profile.getName());

        return rowView;
    }
}

File: aosp/src/apps/VehicleEqualizerFull/app/src/main/java/com/leandromendes/vehicleequalizer/util/ProfileListUtils.kt
package com.leandromendes.vehicleequalizer.util

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.leandromendes.vehicleequalizer.R
import com.leandromendes.vehicleequalizer.data.model.EqualizerProfile

class ProfileListUtils(
    private val context: Context,
    private val profiles: MutableList<EqualizerProfile>
) : ArrayAdapter<EqualizerProfile?>(
    context, android.R.layout.simple_list_item_2,
    profiles
) {
    // Overwrites the main method to create and return the View of an item in the list
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val rowView = inflater.inflate(R.layout.item_profile, parent, false)

        val profileName = rowView.findViewById<TextView>(R.id.profileName)
        val profile = profiles[position]
        profileName.text = profile.name

        return rowView
    }
}